test/unit/test_ranges.py

```diff
@@ -0,0 +1,310 @@
+# Copyright (c) 2015 Scality
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import base64
+import contextlib
+import errno
+import gzip
+import httplib
+import json
+import os
+import os.path
+import pickle
+import shutil
+import sys
+import tempfile
+import time
+import unittest
+
+import eventlet
+
+import swift_scality_backend.server
+
+import swift.account.server
+import swift.container.server
+import swift.common.ring
+import swift.common.utils
+import swift.proxy.server
+
+try:
+    # Pre storage-policy compatibility
+    import swift.common.storage_policy
+except ImportError:
+    pass
+
+
+class TestRangeHeaders(unittest.TestCase):
+    """
+    Test outgoing range headers to sproxyd from the object server on a partial
+    get of an object. See
+    https://github.com/scality/ScalitySproxydSwift/issues/121 for more details.
+    """
+
+    def setUp(self):
```

Comment:
I believe this one should set `self.sproxyd_request_headers` to an empty dict, otherwise the result from a previous test-case could be re-used (if `sproxyd_mock` doesn't set it for some reason).